pkg/apis/core/types.go

```diff
@@ -486,6 +486,17 @@ const (
 	PersistentVolumeClaimFileSystemResizePending PersistentVolumeClaimConditionType = "FileSystemResizePending"
 )
 
+type PersistentVolumeClaimResizeStatus string
+
+const (
+	PersistentVolumeClaimNoExpansionInProgress         PersistentVolumeClaimResizeStatus = ""
+	PersistentVolumeClaimControllerExpansionInProgress PersistentVolumeClaimResizeStatus = "ControllerExpansionInProgress"
+	PersistentVolumeClaimControllerExpansionFailed     PersistentVolumeClaimResizeStatus = "ControllerExpansionFailed"
+	PersistentVolumeClaimNodeExpansionPending          PersistentVolumeClaimResizeStatus = "NodeExpansionPending"
+	PersistentVolumeClaimNodeExpansionInProgress       PersistentVolumeClaimResizeStatus = "NodeExpansionInProgress"
```

Comment:
I checked the code below, both `NodeExpansionPending` and `NodeExpansionInProgress` have the same handling. Should they be merged?

Reply:
I found this comment in the KEP.  Can you add comments for each status here?  That will help explain what each status means.

```
NodeExpansionPending // state set when external resize controller has finished expanding the volume but further expansion is needed on the node
```

Reply:
Add doc comments.

Reply:
We talked on a call and I can see an argument that having two states is good from the user perspective. Users can see `NodeExpansionPending` when kubelet did not have chance to evaluate the PVC and/or pre-expansion checks failed from some reason. `NodeExpansionInProgress` then means that a volume plugin / CSI driver is really expanding the volume.

I am fine with two separate states then.